Packs/ProofpointThreatProtection/Integrations/ProofpointThreatProtection/ProofpointThreatProtection.py

```diff
@@ -0,0 +1,211 @@
+
+    def get_access_token(self, client_id, client_secret):
+        """
+        Get an access token that was previously created if it is still valid, else, generate a new access token from
+        the API key and secret.
+        """
+        # Check if there is an existing valid access token
+        integration_context = self.get_shared_integration_context()
+        if integration_context.get('access_token') and integration_context.get('expiry_time') > date_to_timestamp(datetime.now()):
+            return integration_context.get('access_token')
+        else:
+            try:
+                res = self._http_request(
+                    method='POST',
+                    full_url=f'{self.get_auth_host()}/token',
+                    headers={'Content-Type': 'application/x-www-form-urlencoded'},
+                    data={
+                        'grant_type': 'client_credentials',
+                        'client_id': client_id,
+                        'client_secret': client_secret
+                    }
+                )
+                if res.get('access_token'):
+                    expiry_time = date_to_timestamp(datetime.now(), date_format='%Y-%m-%dT%H:%M:%S')
+                    expiry_time += res.get('expires_in', 0) * 1000 - 10
+                    context = {
+                        'access_token': res.get('access_token'),
+                        'expiry_time': expiry_time
+                    }
+                    self.set_shared_integration_context(context)
+                    self._headers['Authorization'] = f'Bearer {res.get("access_token")}'
+                    return res.get('access_token')
+
+            except Exception as e:
+                raise (Exception(f'Error occurred while creating an access token. '
+                                 f'Please check the instance configuration.'
+                                 f'\n\n{e.args[0]}'))
+
+    def get_safelist(self, cluster_id):
+        return self._http_request(
+            'GET',
+            url_suffix=URL_SUFFIX_SAFELIST,
+            params={'clusterId': cluster_id}
+        )
+
+    def safelist_add_delete(self, cluster_id, args):
+        import json
+        return self._http_request(
+            'POST',
+            url_suffix=URL_SUFFIX_SAFELIST,
+            headers={'Content-Type': 'application/json'},
+            params={'clusterId': cluster_id},
+            data=json.dumps({
+                'action': args.get('action'),
+                'attribute': args.get('attribute'),
+                'operator': args.get('operator'),
+                'value': args.get('value'),
+                'comment': args.get('comment')
+            })
+        )
+
+    def get_blocklist(self, cluster_id):
+        return self._http_request(
+            'GET',
+            url_suffix=URL_SUFFIX_BLOCKLIST,
+            params={'clusterId': cluster_id}
+        )
+
+    def blocklist_add_delete(self, cluster_id, args):
+        import json
```

Comment:
One could make a strong case for the argument that the json python module is only utilized by the 2 class methods (safelist_add_delete and blocklist_add_delete), providing no functionality for the rest of the code base and therefore could be considered ideally imported only if/when those 2 call-ables are invoked. 📦 All that said ... had I written the bulk and/or original code base for this integration contribution I'm pretty sure I would have globally imported it myself. 

@randomizerxd wrote the original code base and that was handed off to me to facilitate its contribution to XSOAR's marketplace. To make a long story as short as possible ... it wasn't until well after the creation of the FastAPI testing model I employed that that a few minor but siginificant bugs were discovered in the original pull request code base that was handed off to me. 

For each bug that I fixed, I attempted (I say attempted now) to house that work in clean & concise commit that was minimally invasive to the original code base ... not b/c @randomizerxd was particularly fond of his code nor because he cared at all about the integration code ... but rather simply because its akward being the new guy walking in the door and immediately taking someone's already pull-requested code base and hacking it to shreds till it is unrecognizable .... I fear in the end this was the outcome anyway, inspite of my efforts .... anyhow you can reference the specific commit  in question here ...

https://github.com/ahopstetter-sce/content-ProofpointThreatProtection-orig-PR-to-current-commit-base/commit/23fd48c6da42aad4d1492469f11c25e3f2b3d5fa

and as well, that repository has all the cherry-pick'd commits showing the difference from the original pull-request to XSOAR to mine, which may help answer why it turned out that way .... but again I readily admit had I wrote it from the ground up ... yea json would of been a global import with a near 100% certainty on that I'd think ... but stranger things have ...

Reply:
Thank you for that I really enjoyed your explenation 
and of course  I too have these dilemmas when entering a new repo / project.